x/dex/keeper/pool.go

```diff
@@ -177,3 +177,20 @@ func (k Keeper) SetPoolCount(ctx sdk.Context, count uint64) {
 	binary.BigEndian.PutUint64(bz, count)
 	store.Set(byteKey, bz)
 }
+
+func (k Keeper) GetAllPoolShareholders(ctx sdk.Context) map[uint64][]types.PoolShareholder {
+	result := make(map[uint64][]types.PoolShareholder)
+	balances := k.bankKeeper.GetAccountsBalances(ctx)
+	for _, balance := range balances {
+		for _, coin := range balance.Coins {
+			poolID, err := types.ParsePoolIDFromDenom(coin.Denom)
+			if err != nil {
+				continue
```

Comment:
Shouldn't we have a log or something to track errors?

Reply:
Not really an error. We use ParsePoolIDFromDenom to check if the denom is a PoolShare or just another random coin. err in this case just means not a PoolShareDenom. But I can add a function that is a bit more clear.